### PR TITLE
Updating Guzzle and php version

### PR DESCRIPTION
**Pull request by m-timmermann**
- Updated php version to 8.1
- Updated guzzlehttp/guzzle
- Added guzzlehttp/uri-template

### CODE REVIEW

File: src/HttpClient.php [hunk 1]
```diff
@@ -62,8 +62,6 @@ public function execute($method, $urlTemplate, $params = [], $headers = [], $bod
 	// https://github.com/guzzle/guzzle/blob/master/src/Utils.php#L93
 	$url = \GuzzleHttp\UriTemplate\UriTemplate::expand($urlTemplate, $params);
 
-  $url = "https://store.meplato.com/api/v2/catalogs?skip=0&take=0&sort=-created%2Cname";
```
Comment (olivere) [inline src/HttpClient.php]: Wow, really. I did that...? That's not my code 🏃 lol

Comment (m-timmermann) [reply src/HttpClient.php]: No thats on me. I commited it by mistake after changing to guzzlehttp/uri-template

**Review by olivere — APPROVED** [review]
Without knowing too much about current PHP style guides, this seems legit to me.

(Some of) The code you changed, however, is auto-generated and populated by some templates in [the `meplato/apigen` repository](https://github.com/meplato/apigen), with the PHP templates being [here](https://github.com/meplato/apigen/blob/master/php/templates/src/HttpClient.php) to be more specific. It is up to you whether you want to change the generators, or—from now on—handle the code changes manually.

**Comment by iavorJ** [comment]
I wanted to add the current state here, as discussed in the huddle, just so we don't forget where we left it.

There seems to be an issue with the update to PHP 8.1. and PHPUnit. At first I updated to PHP 8.1. and left PHPUnit at `"phpunit/phpunit": "~4.0"`. I got:
```
PHP Fatal error:  Uncaught Error: Call to undefined function each() in /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/src/Util/Getopt.php:38
````
As we discussed, this was to be expected, as PHP >= 8 has no `each()` anymore. Updating to `"phpunit/phpunit": "~9.3"` unfortunately yieds another error:
```
/usr/bin/php /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/phpunit --configuration /home/iavorjelev/code/clients/store2-php-client/phpunit.xml.dist /home/iavorjelev/code/clients/store2-php-client/tests --teamcity --cache-result-file=/home/iavorjelev/code/clients/store2-php-client/.phpunit.result.cache
PHP Fatal error:  Uncaught Error: Class "PHPUnit_Framework_TestCase" not found in /home/iavorjelev/code/clients/store2-php-client/tests/BaseTest.php:21
Stack trace:
#0 /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/src/Util/FileLoader.php(65): include_once()
#1 /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/src/Util/FileLoader.php(49): PHPUnit\Util\FileLoader::load()
#2 /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/src/Framework/TestSuite.php(394): PHPUnit\Util\FileLoader::checkAndLoad()
#3 /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/src/Framework/TestSuite.php(524): PHPUnit\Framework\TestSuite->addTestFile()
#4 /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/src/Runner/BaseTestRunner.php(98): PHPUnit\Framework\TestSuite->addTestFiles()
#5 /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/src/TextUI/Command.php(120): PHPUnit\Runner\BaseTestRunner->getTest()
#6 /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/src/TextUI/Command.php(96): PHPUnit\TextUI\Command->run()
#7 /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/phpunit(98): PHPUnit\TextUI\Command::main()
#8 {main}

Next PHPUnit\TextUI\RuntimeException: Class "PHPUnit_Framework_TestCase" not found in /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/src/TextUI/Command.php:98
Stack trace:
#0 /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/phpunit(98): PHPUnit\TextUI\Command::main()
#1 {main}
  thrown in /home/iavorjelev/code/clients/store2-php-client/vendor/phpunit/phpunit/src/TextUI/Command.php on line 98

Process finished with exit code 255
```

**Review by iavorJ — APPROVED** [review]
LGTM, thanks!

Just one nit: we should probably add some name [here](https://github.com/meplato/store2-php-client/blob/master/phpunit.xml.dist#L4), otherwise we get a warning while running the tests:
```
  Warning - The configuration file did not pass validation!
  The following problems have been detected:

  Line 4:
  - Element 'testsuite': The attribute 'name' is required but missing.
```

**Comment by iavorJ** [comment]
> Without knowing too much about current PHP style guides, this seems legit to me.
> 
> (Some of) The code you changed, however, is auto-generated and populated by some templates in [the `meplato/apigen` repository](https://github.com/meplato/apigen), with the PHP templates being [here](https://github.com/meplato/apigen/blob/master/php/templates/src/HttpClient.php) to be more specific. It is up to you whether you want to change the generators, or—from now on—handle the code changes manually.

@m-timmermann Also, we should probably create an issue to adjust the templates, as this makes the client update easier and less error prone. WDYT?

**Pull request merged**